src/i-configuration.ts

```diff
@@ -131,3 +175,72 @@ export class StoreBackedConfiguration implements IConfiguration {
     return this.banditVariationConfigurationStore?.entries() ?? {};
   }
 }
+
+export class ImmutableConfiguration implements IConfiguration {
+  private readonly flags: Record<FlagKey, Flag | ObfuscatedFlag>;
+  private readonly banditVariations?: Record<FlagKey, BanditVariation[]>;
+  private readonly bandits?: Record<string, BanditParameters>;
+  private readonly flagConfigDetails: ConfigDetails;
+
+  constructor(
+    flags: Record<FlagKey, Flag | ObfuscatedFlag>,
+    private readonly initialized: boolean,
+    private readonly obfuscated: boolean,
+    flagConfigDetails: ConfigDetails,
+    banditVariations?: Record<FlagKey, BanditVariation[]>,
+    bandits?: Record<string, BanditParameters>,
+  ) {
+    this.flags = JSON.parse(JSON.stringify(flags));
+    this.flagConfigDetails = JSON.parse(JSON.stringify(flagConfigDetails));
+    this.banditVariations = banditVariations
+      ? JSON.parse(JSON.stringify(banditVariations))
+      : undefined;
+    this.bandits = bandits ? JSON.parse(JSON.stringify(bandits || {})) : undefined;
```

Comment:
If you want to further inforce immutability, you could look into leveraging [`Object.freeze()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze)